trucks.js

```diff
@@ -248,9 +250,27 @@ var foodTrucks = [
   }
 ];
 
-/* 
- * return an object from the module with a single method on it: filterByDay
- * that method should take a single parameter that represents the day to filter on
- * use the built-in filter() method to return all trucks that have the day in their
- * schedule
- */
\ No newline at end of file
+function filterTrucksByDay( day ) {
+
+  var trucks;
+
+  function getTruck( obj ) {
+
+    if ( obj.schedule && obj.schedule.indexOf( day ) > -1 ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  trucks = foodTrucks.filter( getTruck );
+
+  return trucks;
+}
+
+var truckInfo = {
+  foodTrucks : foodTrucks,
```

Comment:
I would avoid returning access to the internal array here. Instead, it should be treated as private data.